### ISSUE TITLE
[eqeqeq] error after update to v18

### ISSUE THREAD

**Issue by DerHerrGammler** [angular-eslint/angular-eslint#1875]
**Description and reproduction of the issue**

Since the update to v18 I got a error from the `eqeqeq` rule. I am not sure if thats a problem of the rule or of the rule or the template parser itself. But here is the verbose stacktrace:

```sh
Cannot destructure property 'source' of '(0 , get_nearest_node_from_1.getNearestNodeFrom)(...)' as it is null.
Occurred while linting C:\workspace\[...].component.html:NaN
Rule: "@angular-eslint/template/eqeqeq"
TypeError: Cannot destructure property 'source' of '(0 , get_nearest_node_from_1.getNearestNodeFrom)(...)' as it is null.
Occurred while linting C:\workspace\[...].component.html:NaN
Rule: "@angular-eslint/template/eqeqeq"
    at getFix (C:\workspace\node_modules\@angular-eslint\eslint-plugin-template\dist\rules\eqeqeq.js:80:13)
    at Object.fix (C:\workspace\node_modules\@angular-eslint\eslint-plugin-template\dist\rules\eqeqeq.js:66:53)
    at normalizeFixes (C:\workspace\node_modules\eslint\lib\linter\report-translator.js:194:28)
    at C:\workspace\node_modules\eslint\lib\linter\report-translator.js:224:22
    at Array.map (<anonymous>)
    at mapSuggestions (C:\workspace\node_modules\eslint\lib\linter\report-translator.js:218:10)
    at C:\workspace\node_modules\eslint\lib\linter\report-translator.js:366:55
    at Object.report (C:\workspace\node_modules\eslint\lib\linter\linter.js:1047:41)
    at Binary[operation=/^(==|!=)$/] (C:\workspace\node_modules\@angular-eslint\eslint-plugin-template\dist\rules\eqeqeq.js:51:25)
    at ruleErrorHandler (C:\workspace\node_modules\eslint\lib\linter\linter.js:1076:28)
```

```JSON
{
  "rules": {
      "@angular-eslint/template/eqeqeq": [
          "error",
          {
              "allowNullOrUndefined": false,
          },
      ],
  }
}
```

I actually got two components where this error occures:  
First:
```HTML
<ng-container *transloco="let t">
    <mat-card>
        <mat-card-header>
            <mat-card-title>
                <span>{{ t("dash.welcome", { email: displayName() }) }}</span>
            </mat-card-title>
        </mat-card-header>
        <mat-card-content>
            <p>
                {{ t("dash.intro") }}
            </p>
        </mat-card-content>
    </mat-card>
    <stepquest-global-statistics [stats]="globalStats()" />
    <form [formGroup]="dailylogForm">
        <mat-card class="mt-4 pb-3">
            <mat-card-header>
                <mat-card-title>
                    {{ t("dailylog.title") }}
                </mat-card-title>
            </mat-card-header>
            <mat-card-content>
                <p tabindex="0" class="info" (click)="openInfoModal()" (keypress)="openInfoModal()">
                    <span>{{ t("howItWorks.title") }}</span>
                    <mat-icon svgIcon="mdi:information-outline" />
                </p>
                @if (noChallengeJoined()) {
                    <mtx-alert type="info"> {{ t("dailylog.noChallenges") }} </mtx-alert>
                }
                <mat-button-toggle-group
                    multiple
                    hideMultipleSelectionIndicator="false"
                    formControlName="collectedStats"
                >
                    <mat-button-toggle value="walk">
                        <mat-icon svgIcon="mdi:walk" />
                    </mat-button-toggle>
                    <mat-button-toggle value="bike">
                        <mat-icon svgIcon="mdi:bike" />
                    </mat-button-toggle>
                    <mat-button-toggle value="swim">
                        <mat-icon svgIcon="mdi:swim" />
                    </mat-button-toggle>
                </mat-button-toggle-group>
                @if (
                    dailylogForm.controls.date.errors?.["dateIsToday"] != null ||
                    dailylogForm.controls.date.errors?.["dateIsFuture"] != null
                ) {
                    <mtx-alert type="warning"> {{ t("dailylog.dateNotInPast") }} </mtx-alert>
                }
                <input-date-time controlName="date" [label]="t('general.date')" />
                @if (dailylogForm.controls.steps.enabled) {
                    <input-number
                        controlName="steps"
                        [label]="t('dailylog.steps')"
                        [hintStart]="
                            dailylogForm.controls.steps.errors?.['isInteger']
                                ? t('error.inputShouldNotFloat')
                                : null
                        "
                    />
                    <input-checkbox controlName="phone" [label]="t('dailylog.phoneTracked')" />
                }
                @if (dailylogForm.controls.bike.enabled) {
                    <input-number controlName="bike" [label]="t('dailylog.bike')" />
                }
                @if (dailylogForm.controls.swim.enabled) {
                    <input-number controlName="swim" [label]="t('dailylog.swim')" />
                }
            </mat-card-content>
            <mat-card-actions>
                <button type="button" mat-stroked-button color="accent" routerLink="/app/dailylog">
                    {{ t("dailylog.view") }}
                </button>
                <button
                    type="submit"
                    mat-flat-button
                    color="primary"
                    [disabled]="dailylogForm.invalid"
                    (click)="submitDailylog()"
                >
                    {{ t("general.submit") }}
                </button>
            </mat-card-actions>
        </mat-card>
    </form>
    <stepquest-challenges-card
        titleKey="challenge.current"
        [challenges]="challenges()"
        (clickJoin)="joinChallenge($event)"
        (clickLeave)="leaveChallenge($event)"
    />
</ng-container>
```
Second:
```html
<input #hiddenFileInput hidden accept="image/*" type="file" (change)="fileChangeEvent($event)" />

<div class="cropArea">
    <div class="cropArea__controls">
        <button type="button" mat-flat-button (click)="rotate(1)">
            <mat-icon svgIcon="mdi:rotate-right" />
            {{ options.translation.rotate }}
        </button>
        <button type="button" mat-flat-button (click)="zoom(0.1)">
            <mat-icon svgIcon="mdi:plus-circle-outline" />
            {{ options.translation.zoom }}
        </button>
        <button type="button" mat-flat-button (click)="zoom(-0.1)">
            <mat-icon svgIcon="mdi:minus-circle-outline" />
            {{ options.translation.zoom }}
        </button>
    </div>
    <image-cropper
        #cropper
        class="cropArea__cropper"
        format="png"
        [imageChangedEvent]="imageChangedEvent()"
        [maintainAspectRatio]="true"
        [aspectRatio]="options.aspectRatio ?? 4 / 3"
        [canvasRotation]="canvasRotation()"
        [allowMoveImage]="true"
        [transform]="transform()"
        (transformChange)="transform.set($event)"
        (imageCropped)="croppedEvent.set($event)"
    />
    <div class="cropArea__preview">
        <h4>{{ options.translation.preview }}</h4>
        <!-- since this is local it can be src -->
        <!-- eslint-disable-next-line @angular-eslint/template/prefer-ngsrc -->
        <img alt="" [src]="croppedImageUrl()" />
    </div>
    <div class="copArea__finish">
        <button type="button" mat-flat-button (click)="dialogRef.close()">
            {{ options.translation.cancel }}
        </button>
        <button
            type="button"
            mat-flat-button
            color="primary"
            [disabled]="croppedEvent()?.blob == null"
            (click)="saveImage()"
        >
            {{ options.translation.save }}
        </button>
    </div>
</div>

```

**Versions**

| package                                   | version |
| ----------------------------------        | ------- |
| `@angular-eslint/eslint-plugin-template`  | `18.0.1` |
| `@angular-eslint/template-parser`         | `18.0.1` |
| `@typescript-eslint/parser`               | `7.12.0` |
| `ESLint`                                  | `8.57.0` |
| `node`                                    | `18.20.1` |

```sh
# Please run `npx ng version` in your project and paste the full output here:

     _                      _                 ____ _     ___
    / \   _ __   __ _ _   _| | __ _ _ __     / ___| |   |_ _|
   / △ \ | '_ \ / _` | | | | |/ _` | '__|   | |   | |    | |
  / ___ \| | | | (_| | |_| | | (_| | |      | |___| |___ | |
 /_/   \_\_| |_|\__, |\__,_|_|\__,_|_|       \____|_____|___|
                |___/


Angular CLI: 18.0.0
Node: 18.20.1
Package Manager: npm 10.5.0
OS: win32 x64

Angular: 18.0.2
... animations, cdk, common, compiler, compiler-cli, core, forms
... language-service, material, material-luxon-adapter
... platform-browser, platform-browser-dynamic, platform-server
... router, service-worker

Package                         Version
---------------------------------------------------------
@angular-devkit/architect       0.1800.3
@angular-devkit/build-angular   18.0.3
@angular-devkit/core            18.0.3
@angular-devkit/schematics      18.0.3
@angular/cli                    18.0.0
@angular/pwa                    18.0.3
@schematics/angular             18.0.3
ng-packagr                      18.0.0
rxjs                            7.8.1
typescript                      5.4.5
zone.js                         0.14.6
```

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest supported version of the packages and checked my `ng version` output per the instructions given here.

**Comment by wiegell** [comment]
I get an identical error on:
```
    "@angular-eslint/eslint-plugin": "17.0.1",
    "@angular-eslint/eslint-plugin-template": "^17.2.1",
    "@angular-eslint/template-parser": "17.0.1",
```

My error seems to be caused by:
```
(item | resultValidationFromReference) == ResultValidation.TooLow
```

Like this:
```
<span
  class="biochemistry-value"
  [ngClass]="{
    'too-low': (item | resultValidationFromReference) == ResultValidation.TooLow,
    'too-high': (item | resultValidationFromReference) == ResultValidation.TooHigh
  }"
>
```

**Comment by JamesHenry** [comment]
I'm very sorry for the delay on this @DerHerrGammler, I am not able to reproduce this. 

If you have chance to provide a minimal reproduction repo I will happily take a look, it's always the best and quickest way as it helps me to help you. You could, for example, use https://stackblitz.com or similar for this, to avoid dealing with git repos entirely.

If for every issue/possible issue I am the one that has to take the time (personal, unpaid time) to turn snippets into runnable code, far fewer issues overall can get addressed by me.

I will close this one for now until a reproduction is available.

Thanks again!

**Comment by Conroy-X** [comment]
> I get an identical error on:
> 
> ```
>     "@angular-eslint/eslint-plugin": "17.0.1",
>     "@angular-eslint/eslint-plugin-template": "^17.2.1",
>     "@angular-eslint/template-parser": "17.0.1",
> ```
> 
> My error seems to be caused by:
> 
> ```
> (item | resultValidationFromReference) == ResultValidation.TooLow
> ```
> 
> Like this:
> 
> ```
> <span
>   class="biochemistry-value"
>   [ngClass]="{
>     'too-low': (item | resultValidationFromReference) == ResultValidation.TooLow,
>     'too-high': (item | resultValidationFromReference) == ResultValidation.TooHigh
>   }"
> >
> ```
> 
> EDIT: The error can be solved by upgrading to v18, but that's not ideal, since we are on @angular/cli~17 EDITEDIT: Ok i was on fixed old 17, solved also by going to `^17`



I got the same problem, which seems to be caused by `= ='
eslint-plugin-template in 17.5.3

**Comment by PetervdWal** [comment]
Get a similar error on

`
    "@angular-eslint/eslint-plugin-template": "^19.2.1",
`
With angular template recommended config

Workaround is to change == -> ===.

Expected result: a linting error that == is not correct.

**Comment by PooSham** [comment]
I got this error with the update to Angular 20. I solved it by downgrading to 20.1.*:

```
npm i -D angular-eslint@~20.1.0
```

@JamesHenry Will you please take a look into this again?

**Comment by JamesHenry** [comment]
As I wrote before, please help me to help you:

> If you have chance to provide a minimal reproduction repo I will happily take a look, it's always the best and quickest way as it helps me to help you. You could, for example, use https://stackblitz.com/ or similar for this, to avoid dealing with git repos entirely.

> If for every issue/possible issue I am the one that has to take the time (personal, unpaid time) to turn snippets into runnable code, far fewer issues overall can get addressed by me.

> I will close this one for now until a reproduction is available.

https://github.com/angular-eslint/angular-eslint/issues/1875#issuecomment-2426502549

In general this issue has been closed for a long time, and opened even longer ago. The context changes and this is multiple major versions old and clearly not a persistent issue.

If a reproduction can be provided I will gladly take a look, but that should all be done with fresh and _complete_ context on a new issue please.
